valhalla/sif/dynamiccost.h

```diff
@@ -582,12 +585,21 @@ class DynamicCost {
   virtual bool bicycle() const;
 
   /**
-   * Returns a function/functor to be used in location searching which will
+   * Returns a value between 0 and 1 indicating how
+   * desirable the edge is for use as a location. A value of 0 indicates the
+   * edge is not usable (no access for the travel mode used by this costing)
+   * while 1 indicates the edge is highly preferred. Values in between can be
+   * used to rank edges such that desirable edges that might be slightly
+   * farther from the location than a less desirable edge can be chosen.
+   *
+   * Function to be used in location searching which will
    * exclude and allow ranking results from the search by looking at each
    * edges attribution and suitability for use as a location by the travel
    * mode used by the costing method.
    */
-  virtual const EdgeFilter GetEdgeFilter() const = 0;
+  virtual float Filter(const baldr::DirectedEdge* edge,
```

Comment:
we should change the node filter into a function on the costing object rather than a functor as well. the reason that they werent changed to this in the past was just that the interface was different long ago. this is much better!